tianshou/env/venvs.py

```diff
@@ -195,15 +196,26 @@ def reset(
         id = self._wrap_id(id)
         if self.is_async:
             self._assert_id(id)
+
         # send(None) == reset() in worker
         for i in id:
-            self.workers[i].send(None)
-        obs_list = [self.workers[i].recv() for i in id]
+            self.workers[i].send(None, **kwargs)
+        ret_list = [self.workers[i].recv() for i in id]
+
+        if "return_info" in kwargs and kwargs["return_info"]:
+            obs_list = [r[0] for r in ret_list]
+        else:
+            obs_list = ret_list
```

Comment:
Is it possible to check it by type?
```python
if isinstance(ret_list[0], tuple) and len(ret_list[0]) == 2:
  # return obs, info
# return obs
```

I know it's a little bit confusing since the observation may also be a tuple. However, I personally don't like this gym's API. Usually the user won't change the environment return data type during the whole process, so the `return_info` argument should be placed in `__init__` instead of `reset` function.

From this point of view, envpool uses `gym_reset_return_info` option in initialization. However, this method cannot support envpool. If possible, would you please add a test case for envpool? Though it only affects the venv wrapper part.

Reply:
Also, we don't support tuple observation space and recommend using dict space. This is quite a strong assumption you can use here. If anyone uses tuple observation and meet exception, we can raise the corresponding hint for them by saying something like `please change your observation space from tuple to array or dict space`

Reply:
I think checking by type is a workable approach, adopted it in 364b46e249f70b21dd8c3b46dcdcb24fe1162b88 . 
Also added test for envpool, and added exception for using tuple observation in the same commit